### PR TITLE
chore: script to apply versions.txt to pom.xml

### LINKED ISSUE #8532: OwlBot is modifying versions that should be managed by Release Please
OwlBot is modifying versions of the artifacts in BOMs. These values should be managed by Release Please.

Screen Shot 2022-10-05 at 3 28 24 PM

Maybe it's no harm so far, as long as Release Please sets the versions in release pull requests.

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #8532 with `generation/apply_current_versions.sh`.

### CODE REVIEW

File: generation/apply_current_versions.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+function apply_module_versions_file() {
+  versions_file=$1
+
+  # Normally java-XXX. For the root pom.xml, it's "."
+  MODULE=$(dirname $versions_file)
+
+  for KV in $(cut -f1,3 -d: $versions_file |grep -v "#"); do
+    K=${KV%:*}; V=${KV#*:}
+    echo Key:$K, Value:$V;
+    pom_files="$(find $MODULE -maxdepth 3 -name pom.xml) pom.xml CoverageAggregator/pom.xml google-cloud-gapic-bom/pom.xml"
```
Comment (meltsufin) [inline generation/apply_current_versions.sh:21]: Why do you have to list `CoverageAggregator` and `google-cloud-gapic-bom` explicitly? Wouldn't those poms be found automatically?

**Review by meltsufin — APPROVED** [review]

**Review by meltsufin — COMMENTED** [review]
Also, do you want to add this to `bootstrap.sh` and the `generated_files_sync.yaml` checks?

File: generation/apply_current_versions.sh (hunk 1 above)
Comment (suztomo) [reply generation/apply_current_versions.sh:21]: It's finding pom.xml within the `$MODULE` directory. It does not try to find other 100 pom.xml files in other modules.

Comment (meltsufin) [reply generation/apply_current_versions.sh:21]: Wouldn't it be more thorough, if it did? Just scan all pom.xml's for the version ref.

Comment (suztomo) [reply generation/apply_current_versions.sh:21]: It's unnecessarily slow. If you have a module that references a version in another module, then it makes sense. Do you know such module?

**Comment by suztomo** [comment]
@meltsufin Added this script to bootstrap.sh and added steps in generated_files_sync.yaml checks.

**Comment by suztomo** [comment]
Bootstrap failed. 'remote: Permission to googleapis/google-cloud-java.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/googleapis/google-cloud-java.git/': The requested URL returned error: 403'. This seems irrelevant.

**Comment by suztomo** [comment]
```
[ERROR] Errors: 
[ERROR] com.google.cloud.dataproc.v1.it.ITSystemTest.null
[ERROR]   Run 1: ITSystemTest.beforeClass:79 » Execution com.google.api.gax.rpc.InvalidArgumentException: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: Insufficient 'CPUS' quota. Requested 12.0, available 2.0.
[ERROR]   Run 2: ITSystemTest.afterClass:93 NullPointer
```

File: generation/apply_current_versions.sh (hunk 1 above)
Comment (meltsufin) [reply generation/apply_current_versions.sh:21]: It seems to be the case in at least `java-asset`.

**Pull request merged**